### PR TITLE
Switch to 'call func'.

### PR DESCRIPTION
**Pull request by rxwei**
Before:
`func applied(to input: Input) -> Output`
After:
`call func(_ input: Input) -> Output`

### CODE REVIEW

File: Tests/DeepLearningTests/TrivialModelTests.swift [hunk 1]
```diff
@@ -34,7 +34,7 @@ final class TrivialModelTests: XCTestCase {
                 )
             }
             @differentiable
-            func applied(to input: Tensor<Float>) -> Tensor<Float> {
+            call func(_ input: Tensor<Float>) -> Tensor<Float> {
                 let h1 = l1.applied(to: input)
```
Comment (dan-zheng) [inline Tests/DeepLearningTests/TrivialModelTests.swift]: What about call-sites?
```suggestion
                let h1 = l1(input)
```

Comment (rxwei) [reply Tests/DeepLearningTests/TrivialModelTests.swift]: Sure. Still building a toolchain.

**Comment by rxwei** [comment]
The existing implementation of callables seems to have a name lookup issue cross-module. If you run `swift test` using the latest toolchain, you'll get these lookup errors:

```console
/Users/rxwei/Development/ML/swift-apis/Tests/DeepLearningTests/ContextTests.swift:24:31: error: cannot call value of non-function type 'Dropout<Float>'
        XCTAssertEqual(dropout(x), x)
                       ~~~~~~~^
```

**Comment by dan-zheng** [comment]
Verified that all tests pass after https://github.com/apple/swift/pull/24121 is merged.
(I built a toolchain based on https://github.com/apple/swift/pull/24121 and ran `swift test`).

**Pull request merged**